torch/_inductor/cudagraph_trees.py

```diff
@@ -146,6 +146,19 @@ def clear_cublas_manager():
         clear_cublass_cache()
 
 
+@contextlib.contextmanager
+def enable_history_recording():
+    "Turns on history recording in the CUDA Caching Allocator"
+    enabled = torch._C._cuda_isHistoryEnabled()
```

Comment:
There are caveats when simple recording is enabled but stack traces are not. This function take like 4 arguments, I wonder if when `_record_memory_history` is called, can we return the current settings as a dict. Maybe for now, we can just have `_record_memory_history` return whether it was enabled before. I guess I am trying to avoid having to add more virtual functions to the allocator interface than necessary.